src/lib.rs

```diff
@@ -7,6 +7,8 @@ use wasm_bindgen::prelude::*;
 
 #[macro_use]
 mod context;
+#[cfg(feature = "editorconfig")]
+pub mod editorconfig;
```

Comment:
Full public visibility to editorconfig is probably not ideal, since it shouldn't be an API consumers rely on. I don't think there is much we can do about it though, `pub(crate)` is probably too restrictive.

Reply:
Indeed, there are [ways](https://users.rust-lang.org/t/pub-only-for-tests/28153) around it but it seems more involved.

I just pushed some cleanup, hopefully fixing the case where the input file would be left modified if `test_stylua_toml` failed, and removing the EOL test attempt (too tricky).

My last concern would be `MaxLineLen::Off => config.with_column_width(0)` since I did not test if `0 === off`.

Reply:
`max_line_length = off` was not working, fixed in the latest commit using `usize::MAX` instead of `0`.